testing/esp/esdlcmd/inputs/ws_exc_test_1.ecm

```diff
@@ -42,6 +42,7 @@ ESPservice [
     auth_feature("DEFERRED"),
     version("1"),
     default_client_version("1"),
+    service_use_latest_version,
```

Comment:
@afishbeck Isn't default_client_version used for all clients including 3rd party clients?  If so, could the default_client_version set to one version whilst internal clients could set to use the latest version?

Reply:
That is true as coded in this PR, but not sure correct/desirable.
I think it should 1st check for the new flag, and return 'version' if present, and fallback to looking at 'default_client_version' if new flag isn't present.

Given this is a service setting for generated code clients only, and I think we said in time we want it to become the default, I think it should override default_client_version.
i.e. external clients, will continue to supply their own version and fallback to 'default_client_version' is none sent.
internal code generated client code will use always use 'version' if this new flag is present.

There no cases where default_client_version is present and != version.
So if this was flag was everywhere (or on by default implicitly), then the only code generated services it would effect as you've noted in the JIRA would be : ws_access, WsResources, FileSpray, WsESDLConfig,
WsCloud, ws_machine, WsResources and WsTopology.

There doesn't seem any point in putting on every ECM. If we want to make this behaviour the default now, we can assume it's true, and code hidlcomp accordingly (no need to add the attribute).

But I think we said we would only enable in ws_fs.ecm in this initial phase.
Then make default everywhere (and remove attribute), but only after testing it doesn't break the effected other services (WsResources, FileSpray, WsESDLConfig,
WsCloud, ws_machine, WsResources and WsTopology).

Reply:
Hmmmm. Ok remembering what we are trying to accomplish here now.

The truth is the way default_client_version is currently used makes no sense.  The value for the server side default and the generated client aren't related and shouldn't be tied like that.  The same tag used both places is probably why it had to be updated constantly. 

The generated client SHOULD default to latest version, I'm just not sure if there is risk of something breaking.  It's generally our own services and tools that use the generated code.

Perhaps reverse the logic?  
1. Generated client never uses default_client_version.  That's for server side.
2. Default the generated client version to latest.
3. For any service where default_client_version is currently set to latest, do nothing, it's going to work.
4. If there are generated clients that might break... add a new tag generated_client_version("1.17") to make it match what it is now.  
5. Otherwise let it default to latest.  
 
And at then we've decoupled the server side default from the client side one.
No matter what don't use the name service_use_latest_version it sounds like a server side setting.

Reply:
1,2,3 - I think we are essentially saying the same thing (if the new semantics were true by default).

4 - >to make it match what it is now.

What it is now, is version=0 for the effected services.
i.e. ws_access, WsResources, FileSpray, WsESDLConfig, WsCloud, ws_machine, WsResources and WsTopology
Currently sends no version, and so the server treats them as version 0 (all min_ver fields are stripped).

To keep these services behaving as they do now with 'generated_client_version' would mean using generated_client_version(0.0) I think.
@afishbeck - can you confirm that is what you meant?

In this scheme ws_fs.ecm would not require any changes, it would like most others just use 'version'.

Either way:
1) This is a temporary step, until we can prove the other affected client services are okay (as they should be) using the latest version.
2)  it means either:
      I)  Add "use_version_for_internal_clients(1)" to ws_fs.ecm
OR ii) Add "generated_client_version(0.0)" to WsESDLConfig, WsCloud, ws_machine, WsResources and WsTopology

Reply:
You've convinced me that a better route too - add generated_client_version(0.0) to the services that need them only.

>Some of the services you mentioned probably don't even have clients using the generated client code.

Yes, and for those, using 'version' can't do any harm, since there are no use cases.
>ws_access, WsResources, FileSpray, WsESDLConfig, WsCloud, ws_machine, WsResources and WsTopology
i.e. it is not worth adding generated_client_version(0.0) for any of these that do not have any code that uses them.